beets/util/artresizer.py

```diff
@@ -309,10 +333,10 @@ def compare(self, im1, im2, compare_threshold):
         return phash_diff <= compare_threshold
 
     @property
-    def can_write_metadata(self):
+    def can_write_metadata(self) -> bool:
         return True
 
-    def write_metadata(self, file, metadata):
+    def write_metadata(self, file: AnyStr, metadata: Mapping):
         assignments = list(chain.from_iterable(
```

Comment:
Looks like something about this makes it "untypeable", I think we basically have no choice but to explicitly type it as `Any`...

```
beets/util/artresizer.py:344: error: Cannot determine type of "assignments"  [has-type]
```

Doesn't seem liek too big of a shortcut to me given that `util.command_output` is untyped anyway.

Reply:
If you're using mypy to get these hints, then you should know that I have been working under the understanding that mypy is not really relevant at this stage. It is much more strict and even without this PR, throws a huge number of errors. See [here for details](https://github.com/beetbox/beets/pull/4495#issuecomment-1328068170). 

That MyPy gives this error doesn't mean it's true. It doesn't mean that it is actually `Any`. Those are the typings that are presented by the codebase and those ones I'm sure of.

Reply:
Well, I wouldn't say mypy is completely irrelevant, it can definitely provide useful input.

What is certainly the case that having zero mypy issues left is not a requirement for a PR like this to be merged, since that might simply not be achievable without substantial refactoring to resolve issues that mypy reveals.

An example would be the `version()` problem mentioned in another comment: The refactoring proposed by @ybnd looks like a great idea, but I also agree that we might want to hold it off for another PR in order to avoid blowing up this one too much (although, in this case, I'd be fine either way: the PR is not that huge yet).

---

As to the particular issue, I'm somewhat surprised that mypy cannot resolve this to at least a `List[Any]`, which would be enough to verify that the list unpacking in the argument list is valid. Maybe explicitly annotating it as `List[Any]` would be useful? Or, maybe specifying the `Mapping` in the argument more concretely. I suspect this should be `Mapping[str, Any]`, maybe even `Mapping[str, str]` (but verifying the latter would require some research into the backend methods).